src/scripting_api/util.js

```diff
@@ -20,11 +20,734 @@ class Util extends PDFObject {
     super(data);
 
     this._crackURL = data.crackURL;
+    this._scandCache = new Map();
+    this._months = [
+      "January",
+      "February",
+      "March",
+      "April",
+      "May",
+      "June",
+      "July",
+      "August",
+      "September",
+      "October",
+      "November",
+      "December",
+    ];
+    this._days = [
+      "Sunday",
+      "Monday",
+      "Tuesday",
+      "Wednesday",
+      "Thursday",
+      "Friday",
+      "Saturday",
+    ];
   }
 
   crackURL(cURL) {
     return this._crackURL(cURL);
   }
+
+  printf(...args) {
```

Comment:
Is the PDF's printf different than c's printf? It doesn't look like below handles escaping. For example in c `printf("%%s")` should print `%s`

Reply:
`printf("%%s", "foo")` and `printf("%%s")` both print `%%s` in Acrobat.
So I must change my approach.

Reply:
I think there are still some issues:
`%%%%s` should be `%%s`
or `"%%%s", "a"` should be `%a`

IIRC, printf format is not a regular grammar so a regex can't be used to parse it.

Reply:
Did you try it in acrobat ?
According to the spec, only `printf` function is pretty basic:
https://www.adobe.com/content/dam/acom/en/devnet/acrobat/pdfs/js_api_reference.pdf#page=716
or I missed something

Reply:
Tried in acrobat, you're right. I was going off of how printf in c actually works. I guess it's not the same cFormat.